datasets/esa-cci-lc/esa_cci_lc.py

```diff
@@ -0,0 +1,68 @@
+from tempfile import TemporaryDirectory
+from typing import List, Union
+from pathlib import Path
+
+import pystac
+from stactools.esa_cci_lc.cog.stac import create_items as cog_create_items
+from stactools.esa_cci_lc.netcdf.stac import create_item as netcdf_create_item
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+
+class EsaCciLcCog(Collection):
+    @classmethod
+    def create_item(
+        cls, asset_uri: str, storage_factory: StorageFactory
+    ) -> Union[List[pystac.Item], WaitTaskResult]:
+
+        with TemporaryDirectory() as tmp_dir:
+            storage, nc_path = storage_factory.get_storage_for_file(asset_uri)
+            tmp_nc_path = str(Path(tmp_dir, Path(nc_path).name))
+            storage.download_file(nc_path, tmp_nc_path)
+
+            items = cog_create_items(
+                nc_path=tmp_nc_path,
+                cog_dir=tmp_dir,
+                nc_api_url=(
```

Comment:
(Making an item locally now to confirm, but writing this first)

I this set as some kind of "derived_from" link? I *think* we have something special in our API for this. I'm pretty sure that if you have a path here like `esa-cci-lc-netcdf/items/{item_id}` (substituting for the real item ID), our STAC API server will dynamically transform that to `https://planetarycomputer.microsoft.com/api/stac/v1/collections/esa-cci-lc-netcdf/items/{item_id}`

Reply:
I see that it is a derived from: https://github.com/stactools-packages/esa-cci-lc/blob/de196366c565e4e2eff63b63be3a4cb609f54390/src/stactools/esa_cci_lc/cog/stac.py#L122-L131

We have an example of this for sentinel-1-rtc derived from sentinel-1-grd. See https://github.com/microsoft/planetary-computer-tasks/blob/8586b72d6880cd312065d782109e10efa72c8127/datasets/sentinel-1-rtc/s1rtc.py#L82-L93.

So you'd want `collections/esa-cci-lc-netcdf/items/<netcdf-item-id>`.

Reply:
Sounds good. Updated.